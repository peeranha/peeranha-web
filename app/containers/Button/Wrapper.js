import styled from 'styled-components';

const Wrapper = styled.div`
  padding: 40px 20px;
  .border-top-2 {
    border-top: 2px solid rgba(0, 0, 0, 0.2);
  }
  p {
    text-align: center;
  }
  a,
  .btn-link {
    color: #000;
  }
  .border-or-top-2::before,
  .border-or-top-2::after {
    content: '';
    width: 40%;
    height: 1px;
    margin: 3px 10px;
    display: inline-block;
    background: #000;
  }
`;

export default Wrapper;
