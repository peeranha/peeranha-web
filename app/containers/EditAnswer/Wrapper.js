import React from 'react';
import PropTypes from 'prop-types';

import * as routes from 'routes-config';
import commonMessages from 'common-messages';
import { FormattedMessage } from 'react-intl';
import { TEXT_PRIMARY } from 'style-constants';

import myFeedIcon from 'images/myFeed.svg';
import closeIcon from 'images/closeCircle.svg';

import { MediumImageStyled } from 'components/Img/MediumImage';
import BaseRounded from 'components/Base/BaseRounded';
import Tips from 'components/TextEditor/Tips';
import Span from 'components/Span';
import H3 from 'components/H3';
import A from 'components/A';

import messages from './messages';

const Wrapper = /* istanbul ignore next */ ({
  children,
  questionid,
  answerid,
}) => (
  <div>
    <BaseRounded className="d-flex align-items-center justify-content-between mb-3">
      <H3 className="d-flex align-items-end">
        <MediumImageStyled src={myFeedIcon} alt="edit-answer-icon" />
        <FormattedMessage {...messages.editAnswer} />
      </H3>

      <A
        to={routes.questionView(questionid, answerid)}
        href={routes.questionView(questionid, answerid)}
        className="d-inline-flex align-items-center"
      >
        <img className="mr-1" src={closeIcon} alt="x" />
        <Span color={TEXT_PRIMARY}>
          <FormattedMessage {...commonMessages.close} />
        </Span>
      </A>
    </BaseRounded>

    <BaseRounded className="p-0">
      <div className="d-flex">
        <div className="col-12 col-xl-9 p-0">
          <BaseRounded>{children}</BaseRounded>
        </div>

        <Tips className="d-none d-xl-block col-xl-3 p-0" />
      </div>
    </BaseRounded>
  </div>
);

Wrapper.propTypes = {
  children: PropTypes.any,
  questionid: PropTypes.string,
  answerid: PropTypes.string,
};

export default React.memo(Wrapper);
