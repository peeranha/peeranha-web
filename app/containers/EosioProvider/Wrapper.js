import styled from 'styled-components';

const LoadingIndicatorCell = `40px`;
const Wrapper = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  > div {
    width: ${LoadingIndicatorCell};
    height: ${LoadingIndicatorCell};
    position: absolute;
    top: calc(45vh - ${LoadingIndicatorCell});
  }
`;

export default Wrapper;
