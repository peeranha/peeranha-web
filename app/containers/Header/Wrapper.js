import styled from 'styled-components';
import { BG_LIGHT } from 'style-constants';

import { HEADER_HEIGHT } from './constants';

const Wrapper = styled.header`
  display: flex;
  align-items: center;
  position: fixed;
  height: ${HEADER_HEIGHT}px;
  z-index: 9999;
  background: ${BG_LIGHT};
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.19);
  padding: 20px 0;
  width: 100%;
  transition: 0.4s;

  &.scroll-hidden {
    top: 0px;
    transform: translate(0px, ${-HEADER_HEIGHT}px);
    box-shadow: none;
  }

  &.scroll-visible {
    top: ${-HEADER_HEIGHT}px;
    transform: translate(0px, ${HEADER_HEIGHT}px);
  }
`;

export default Wrapper;
