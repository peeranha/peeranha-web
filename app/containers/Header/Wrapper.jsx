import styled from 'styled-components';
import { BG_LIGHT, BORDER_SECONDARY } from 'style-constants';

import { HEADER_HEIGHT, MOBILE_HEADER_HEIGHT } from './constants';

const Wrapper = styled.header`
  display: flex;
  align-items: center;
  position: fixed;
  height: ${HEADER_HEIGHT}px;
  transform: translate(0px, 0px);
  z-index: 9999;
  background: ${BG_LIGHT};
  box-shadow: 0 2px 4px 0 ${BORDER_SECONDARY};
  padding: 20px 0;
  width: 100%;
  transition: all 0.4s ease;

  form {
    flex: 1;
  }

  &.sticky {
    transform: translate(0px, -${HEADER_HEIGHT}px);
  }

  @media only screen and (max-width: 576px) {
    height: ${MOBILE_HEADER_HEIGHT}px;
    box-shadow: 0 1px 2px 0 ${BORDER_SECONDARY};

    &.sticky {
      transform: translate(0px, -${MOBILE_HEADER_HEIGHT}px);
    }
  }
`;

export default Wrapper;
