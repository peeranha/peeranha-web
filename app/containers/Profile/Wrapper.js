import styled from 'styled-components';

export const AvatarEditorBorder = 20;
export const AvatarEditorBorderRadius = 100;
export const AvatarEditorSize = 160;
export const AvatarEditorColor = [255, 255, 255, 0.6];
export const AvatarEditorScale = 1.5;
export const AvatarEditorRotate = 0;
export const AvatarCrossOrigin = 'anonymous';

const FormWidth = 480;

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 25px 0;
  form,
  .view-form {
    flex-basis: ${FormWidth}px;
    * {
      margin: 3px 0;
    }
  }
  .profile-image {
    width: ${AvatarEditorSize}px;
    height: ${AvatarEditorSize}px;
    border-radius: 50%;
    margin-bottom: 10px;
  }
  .text-danger {
    padding-left: 10px;
    padding-top: 2px;
    font-size: 14px;
  }
  .view-form {
    > * {
      margin: 10px 0;
    }
    p {
      padding: 10px;
      background: rgba(0, 0, 0, 0.05);
    }
  }
`;

export default Wrapper;
