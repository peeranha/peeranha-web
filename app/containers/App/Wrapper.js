import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import $ from 'jquery';

import Header from 'containers/Header';
import LeftMenu from 'containers/LeftMenu';
import Loader from 'components/LoadingIndicator/WidthCentered';

import {
  HEADER_HEIGHT,
  MOBILE_HEADER_HEIGHT,
} from 'containers/Header/constants';

import { LEFT_MENU_WIDTH } from './constants';

const Main = styled.div`
  background: rgb(234, 236, 244);
  min-height: 100vh;
  padding-top: ${x => (!x.isMenuVisible ? HEADER_HEIGHT : 0)}px;
  padding-bottom: ${x => (!x.isMenuVisible ? 75 : 0)}px;

  @media only screen and (max-width: 576px) {
    padding-top: ${x => (!x.isMenuVisible ? MOBILE_HEADER_HEIGHT : 0)}px;
    padding-bottom: 0px;
  }
`;

const WrapStyled = styled.main`
  margin-top: 10px;
  width: calc(100% - ${LEFT_MENU_WIDTH}px);
  overflow-y: -webkit-paged-y;

  @media only screen and (max-width: 992px) {
    width: 100%;
  }

  @media only screen and (max-width: 576px) {
    margin-top: 0px;
  }
`;

export class Box extends React.PureComponent {
  state = {
    isMenuVisible: false,
  };

  // set default state if window resizing is happening
  componentDidMount() {
    $(window).resize(() => {
      clearTimeout(window.resizedFinished);
      window.resizedFinished = setTimeout(() => {
        this.setState({
          isMenuVisible: false,
        });
      }, 250);
    });
  }

  // set default state if links changing is happening
  componentWillReceiveProps(prevProps) {
    if (
      prevProps.props.location.pathname + prevProps.props.location.hash !==
        this.props.props.location.pathname + this.props.props.location.hash &&
      this.state.isMenuVisible
    ) {
      this.setState({
        isMenuVisible: false,
      });
    }
  }

  showMenu = () => {
    this.setState({ isMenuVisible: !this.state.isMenuVisible });
  };

  render() {
    const { isMenuVisible } = this.state;
    const { Comp, props } = this.props;

    return (
      <React.Fragment>
        {!isMenuVisible && (
          <Header
            expandLeftMenuNavigation={this.expandLeftMenuNavigation}
            showMenu={this.showMenu}
          />
        )}

        <Main isMenuVisible={isMenuVisible}>
          <div className={isMenuVisible ? '' : 'container container-mobile'}>
            <div className="d-flex">
              <LeftMenu
                showMenu={this.showMenu}
                isMenuVisible={isMenuVisible}
                {...props}
              />

              <WrapStyled className={`${isMenuVisible ? 'd-none' : ''}`}>
                <React.Suspense fallback={<Loader />}>
                  <Comp {...props} />
                </React.Suspense>
              </WrapStyled>
            </div>
          </div>
        </Main>
      </React.Fragment>
    );
  }
}

Box.propTypes = {
  Comp: PropTypes.any,
  props: PropTypes.object,
};

const Wrapper = (comp, props) => <Box Comp={comp} props={props} />;

export default Wrapper;
