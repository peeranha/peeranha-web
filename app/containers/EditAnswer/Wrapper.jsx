import React from 'react';
import PropTypes from 'prop-types';

import * as routes from 'routes-config';
import commonMessages from 'common-messages';
import { FormattedMessage } from 'react-intl';
import { TEXT_PRIMARY } from 'style-constants';

import myFeedIcon from 'images/myFeedHeader.svg?inline';
import closeIcon from 'images/closeCircle.svg?inline';

import TipsBase from 'components/Base/TipsBase';
import { BaseSpecialOne } from 'components/Base/BaseTransparent';
import { MediumImageStyled } from 'components/Img/MediumImage';
import Header from 'components/Header/Simple';
import Tips from 'components/TextEditor/Tips';
import Span from 'components/Span';
import H3 from 'components/H3';
import A from 'components/A';

import messages from './messages';

const Wrapper = ({ children, questionid, answerid }) => (
  <div>
    <Header className="mb-to-sm-0 mb-from-sm-3">
      <H3>
        <MediumImageStyled src={myFeedIcon} alt="edit-answer-icon" />
        <FormattedMessage {...messages.editAnswer} />
      </H3>

      <div className="right-panel">
        <A to={routes.questionView(questionid, answerid)}>
          <button>
            <img className="mr-1" src={closeIcon} alt="x" />
            <Span color={TEXT_PRIMARY}>
              <FormattedMessage {...commonMessages.close} />
            </Span>
          </button>
        </A>
      </div>
    </Header>

    <TipsBase className="overflow-hidden">
      <BaseSpecialOne>{children}</BaseSpecialOne>
      <Tips />
    </TipsBase>
  </div>
);

Wrapper.propTypes = {
  children: PropTypes.any,
  questionid: PropTypes.string,
  answerid: PropTypes.number,
};

export default React.memo(Wrapper);
