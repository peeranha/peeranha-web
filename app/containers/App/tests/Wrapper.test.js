import React from 'react';
import Wrapper, { Box } from '../Wrapper';

const cmp = new Box();
cmp.props = {
  Comp: () => <div>123</div>,
  props: {},
};

describe('Box', () => {
  it('test', () => {
    expect(cmp.render()).toMatchSnapshot();
  });
});

describe('Wrapper', () => {
  const comp = () => <div>123</div>;
  const props = {};

  it('test', () => {
    expect(Wrapper(comp, props)).toMatchSnapshot();
  });
});
