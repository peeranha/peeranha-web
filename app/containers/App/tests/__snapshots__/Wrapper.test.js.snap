// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box test 1`] = `
<React.Fragment>
  <Connect(Header)
    expandLeftMenuNavigation={[Function]}
    isMenuVisible={false}
    showMenu={[Function]}
  />
  <Wrapper__Main>
    <div
      className="container"
    >
      <div
        className="d-flex"
      >
        <Connect(LeftMenu)
          isMenuVisible={false}
          isNavigationExpanded={false}
          showMenu={[Function]}
        />
        <Wrapper__WrapStyled
          className=""
        >
          <UNDEFINED
            fallback={<UNDEFINED />}
          >
            <Comp />
          </UNDEFINED>
        </Wrapper__WrapStyled>
      </div>
    </div>
  </Wrapper__Main>
</React.Fragment>
`;

exports[`Wrapper test 1`] = `
<Box
  Comp={[Function]}
  props={Object {}}
/>
`;
